File: .codex_review_extract/laravel-api/app/Models/Shift.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;
use Illuminate\Database\Eloquent\Relations\HasOne;

class Shift extends Model
{
    protected $table = 'shifts';
    protected $primaryKey = 'id_shift';
    public $timestamps = true;

    protected $fillable = [
        'id_user',
        'open_time',
        'close_time',
        'petty_cash',
        'status',
    ];

    protected $casts = [
        'petty_cash' => 'decimal:2',
        'open_time' => 'datetime',
        'close_time' => 'datetime',
    ];

    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class, 'id_user', 'id_user');
    }

    public function pettyCashes(): HasMany
    {
        return $this->hasMany(PettyCash::class, 'id_shift', 'id_shift');
    }

    public function transactions(): HasMany
    {
        return $this->hasMany(Transaction::class, 'id_shift', 'id_shift');
    }

    public function expenses(): HasMany
    {
        return $this->hasMany(Expense::class, 'id_shift', 'id_shift');
    }

    public function closing(): HasOne
    {
        return $this->hasOne(Closing::class, 'id_shift', 'id_shift');
    }
}

File: .codex_review_extract/flutter-pos/lib/data/models/shift.dart
import 'num_util.dart';
import 'user.dart';

class Shift {
  final int idShift;
  final int idUser;
  final String? openTime;
  final String? closeTime;
  final double pettyCash;
  final String status;
  final User? user;

  const Shift({
    required this.idShift,
    required this.idUser,
    this.openTime,
    this.closeTime,
    required this.pettyCash,
    required this.status,
    this.user,
  });

  factory Shift.fromJson(Map<String, dynamic> json) => Shift(
        idShift: json['id_shift'] as int,
        idUser: json['id_user'] as int,
        openTime: json['open_time'] as String?,
        closeTime: json['close_time'] as String?,
        pettyCash: parseDouble(json['petty_cash']),
        status: json['status'] as String,
        user: json['user'] != null
            ? User.fromJson(json['user'] as Map<String, dynamic>)
            : null,
      );
}
